src/App.js

```diff
@@ -62,33 +115,25 @@ class App extends Component {
   }
 
   refreshStatus = version => {
-    const stateToUrl = {
-      archive: 'archive',
-      release_notes: 'bedrock/release-notes',
-      security_advisories: 'bedrock/security-advisories',
-      download_links: 'bedrock/download-links',
-      product_details: 'product-details'
+    version = version || this.state.version;
+    if (!version) {
+      return;
     }
-    const updateState = (stateKey, data) => {
-      this.setState(prevState => {
-        prevState.statuses[stateKey] = data
+    fetchStatus(version)
+      .then(statuses => {
+        // Detect if some status changed, and notify!
+        const changed = Object.keys(statuses).map(() => {
+           /* this.state[key] != ...*/
+        });
+        if (changed.length > 0) {
+          notifyChanges(changed);
+        }
+        // Save current state.
+        this.setState({statuses});
```

Comment:
Maybe this could go in the "if changed" part?

Reply:
For the notifications, I omit the case when we pass from null to the first value

Reply:
ah, right, good point